### ISSUE TITLE
NVDA fails to start if Windows is set to an unsupported/impossible locale

### ISSUE THREAD

**Issue by bramd** [nvaccess/nvda#8726]
### Steps to reproduce:

1. Set Windows region/language settings as follows:
  - Region/country: Netherlands
  - Date/time/number format: English
  - Interface language: English
2. Start NVDA

### Actual behavior:

- NVDA fails to start

### Expected behavior:

- NVDA starts normally either with the selected NVDA language or with a fallback to the Windows interface language or English

#### NVDA Installed/portable/running from source:
- Installed

#### NVDA version:
- alpha-15976,8d33d238

#### Windows version:
- Windows 10

#### Does the issue still occur after restarting your PC?
- Yes

#### Have you tried any other versions of NVDA?
- No

**Comment by bramd** [comment]
The problem seems to be that there is no error handling at nvda.pyw line 29, which calls setlocale with an empty string to set it to the system locale. If the system ends up being configured with an impossible setting such as en-nl this fails.

**Comment by leonardder** [comment]
Hmm, I believe I have the same settings (display language and date/time format English UK, country the Netherlands. Still it seems to work fine here.

In the windows 10 language settings, what languages do you have available? Only English, or Dutch as well? Dutch is my default app language, while English is below Dutch in the list and is set as display language.

**Comment by bramd** [comment]
Here, English is the first in the list and set as default app language. 
There are language packs installed for Ducht (Netherlands) and Dutch 
(Belgium).


If you look in the old language settings (Control panel, Region), you 
can select a date/time/number format. This list seems to have way more 
options than it had before. For example, English has variants like 
Europe, Belgium, Netherlands etc. Choosing one of those uncommon 
variants should trigger this. Please note that you might need to restart 
for the changes to take effect.

**Comment by leonardder** [comment]
This looks serious enough for a p2.

**Comment by michaelDCurran** [comment]
I cannot seem to reproduce this. Do we know what exception is being thrown by SetLocale? Also, even if we do catch this,  do other things fail after that point?

**Comment by leonardder** [comment]
I can now reproduce this reliably. It is pretty annoying, so I will take this one.

**Comment by pluma** [comment]
This seems to have been what was preventing me from installing NVDA at all. When running the installer I get "an error has occured" right after the logo and this shows up in the log file (which gets deleted when the dialog is closed, btw, so this took me a while to hunt down):

```
Traceback (most recent call last):
  File "nvda.pyw", line 29, in <module>
  File "locale.pyo", line 581, in setlocale
locale.Error: unsupported locale setting
```

I'm in Germany, Windows is set to "English (Germany)". Switching to "English (United States)" allowed me to run the installer normally.
